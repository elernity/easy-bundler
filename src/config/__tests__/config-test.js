'use strict';
require('babel-register')();

const path = require('path');
const getConfig = require('../config');
const configFilePath = path.join(__dirname, '../__files__/customConfig.json');

describe('配置合并测试', () => {

    it('命令行配置应该覆盖默认配置', () => {
        const shellConfig = {
            projectRoot: '/tmp/root',
            entryFile: '/tmp/root/index.js',
            output: '/tmp/root/output/index.platform.bundle',
            platform: 'ios',
            dev: false,
            assetsOutput: '/tmp/root/output/assets/',
            split: true,
        };
        const config = getConfig(shellConfig);
        expect(config).toMatchSnapshot();
    });

    it('文件配置应该覆盖默认配置', () => {
        const shellConfig = {
            projectRoot: '/tmp/root',
            entryFile: '/tmp/root/index.js',
            output: '/tmp/root/output/index.platform.bundle',
            configFile: configFilePath,
        };
        const config = getConfig(shellConfig);
        delete config.userConfig['configFile'];
        expect(config).toMatchSnapshot();
    });
});
