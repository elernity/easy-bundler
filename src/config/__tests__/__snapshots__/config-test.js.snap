// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`配置合并测试 命令行配置应该覆盖默认配置 1`] = `
Object {
  "systemConfig": Object {
    "assetExts": Array [
      "bmp",
      "gif",
      "jpg",
      "jpeg",
      "png",
      "psd",
      "svg",
      "webp",
      "m4v",
      "mov",
      "mp4",
      "mpeg",
      "mpg",
      "webm",
      "aac",
      "aiff",
      "caf",
      "m4a",
      "mp3",
      "wav",
      "html",
      "pdf",
      "otf",
      "ttf",
    ],
    "createModuleIdFactory": [Function],
    "imageExts": Array [
      "png",
      "jpg",
      "jpeg",
      "bmp",
      "gif",
      "webp",
      "psd",
      "svg",
      "tiff",
    ],
    "mainFields": Array [
      "react-native",
      "browser",
      "main",
    ],
    "minifierConfig": Object {
      "compress": Object {
        "arrows": true,
        "booleans": true,
        "collapse_vars": true,
        "comparisons": true,
        "computed_props": true,
        "conditionals": true,
        "dead_code": true,
        "drop_console": false,
        "drop_debugger": true,
        "ecma": 5,
        "evaluate": true,
        "expression": false,
        "global_defs": Object {},
        "hoist_funs": false,
        "hoist_props": true,
        "hoist_vars": false,
        "ie8": false,
        "if_return": true,
        "inline": 3,
        "join_vars": true,
        "keep_classnames": false,
        "keep_fargs": true,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": true,
        "negate_iife": true,
        "passes": 1,
        "properties": true,
        "pure_funcs": null,
        "pure_getters": "strict",
        "reduce_funcs": false,
        "reduce_vars": true,
        "sequences": true,
        "side_effects": true,
        "switches": true,
        "top_retain": null,
        "toplevel": false,
        "typeofs": true,
        "unsafe": false,
        "unsafe_Function": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_math": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": true,
        "warnings": false,
      },
      "mangle": Object {
        "reserved": Array [
          "g",
          "r",
          "i",
          "a",
          "m",
          "e",
          "d",
        ],
        "toplevel": false,
      },
      "output": Object {
        "ascii_only": true,
        "beautify": false,
        "bracketize": false,
        "comments": false,
        "ecma": 5,
        "ie8": false,
        "indent_level": 4,
        "indent_start": 0,
        "inline_script": true,
        "keep_quoted_props": false,
        "max_line_len": false,
        "preamble": null,
        "preserve_line": false,
        "quote_keys": false,
        "quote_style": 3,
        "safari10": false,
        "semicolons": true,
        "shebang": true,
        "shorthand": false,
        "webkit": false,
        "width": 80,
        "wrap_iife": true,
      },
      "sourceMap": false,
      "toplevel": false,
    },
    "platforms": Array [
      "ios",
      "android",
      "windows",
      "web",
      "native",
    ],
    "runBeforeMainModule": Array [
      "node_modules/react-native/Libraries/Core/InitializeCore.js",
      "node_modules/@sdp.nd/nd-react-wrapper/bundleUtil.js",
    ],
    "searchExts": Array [
      "js",
      "json",
      "ts",
      "tsx",
      "bmp",
      "gif",
      "jpg",
      "jpeg",
      "png",
      "psd",
      "svg",
      "webp",
      "m4v",
      "mov",
      "mp4",
      "mpeg",
      "mpg",
      "webm",
      "aac",
      "aiff",
      "caf",
      "m4a",
      "mp3",
      "wav",
      "html",
      "pdf",
      "otf",
      "ttf",
    ],
    "searchIgnore": [Function],
    "sourceExts": Array [
      "js",
      "json",
      "ts",
      "tsx",
    ],
  },
  "userConfig": Object {
    "assetsOutput": "/tmp/root/output/assets/",
    "configFile": "",
    "dev": false,
    "entryFile": "/tmp/root/index.js",
    "output": "/tmp/root/output/index.platform.bundle",
    "platform": "ios",
    "projectRoot": "/tmp/root",
    "split": true,
  },
}
`;

exports[`配置合并测试 文件配置应该覆盖默认配置 1`] = `
Object {
  "systemConfig": Object {
    "assetExts": Array [
      "bmp",
      "gif",
      "jpg",
      "jpeg",
      "png",
      "psd",
      "svg",
      "webp",
      "m4v",
      "mov",
      "mp4",
      "mpeg",
      "mpg",
      "webm",
      "aac",
      "aiff",
      "caf",
      "m4a",
      "mp3",
      "wav",
      "html",
      "pdf",
      "otf",
      "ttf",
    ],
    "createModuleIdFactory": [Function],
    "imageExts": Array [
      "png",
      "jpg",
      "jpeg",
      "bmp",
      "gif",
      "webp",
      "psd",
      "svg",
      "tiff",
    ],
    "mainFields": Array [
      "react-native",
      "browser",
      "main",
    ],
    "minifierConfig": Object {
      "compress": Object {
        "arrows": true,
        "booleans": true,
        "collapse_vars": true,
        "comparisons": true,
        "computed_props": true,
        "conditionals": true,
        "dead_code": true,
        "drop_console": false,
        "drop_debugger": true,
        "ecma": 5,
        "evaluate": true,
        "expression": false,
        "global_defs": Object {},
        "hoist_funs": false,
        "hoist_props": true,
        "hoist_vars": false,
        "ie8": false,
        "if_return": true,
        "inline": 3,
        "join_vars": true,
        "keep_classnames": false,
        "keep_fargs": true,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": true,
        "negate_iife": true,
        "passes": 1,
        "properties": true,
        "pure_funcs": null,
        "pure_getters": "strict",
        "reduce_funcs": false,
        "reduce_vars": true,
        "sequences": true,
        "side_effects": true,
        "switches": true,
        "top_retain": null,
        "toplevel": false,
        "typeofs": true,
        "unsafe": false,
        "unsafe_Function": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_math": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": true,
        "warnings": false,
      },
      "mangle": Object {
        "reserved": Array [
          "g",
          "r",
          "i",
          "a",
          "m",
          "e",
          "d",
        ],
        "toplevel": false,
      },
      "output": Object {
        "ascii_only": true,
        "beautify": false,
        "bracketize": false,
        "comments": false,
        "ecma": 5,
        "ie8": false,
        "indent_level": 4,
        "indent_start": 0,
        "inline_script": true,
        "keep_quoted_props": false,
        "max_line_len": false,
        "preamble": null,
        "preserve_line": false,
        "quote_keys": false,
        "quote_style": 3,
        "safari10": false,
        "semicolons": true,
        "shebang": true,
        "shorthand": false,
        "webkit": false,
        "width": 80,
        "wrap_iife": true,
      },
      "sourceMap": false,
      "toplevel": false,
    },
    "platforms": Array [
      "ios",
      "android",
      "windows",
      "web",
      "native",
    ],
    "runBeforeMainModule": Array [
      "node_modules/react-native/Libraries/Core/InitializeCore.js",
      "node_modules/@sdp.nd/nd-react-wrapper/bundleUtil.js",
    ],
    "searchExts": Array [
      "js",
      "json",
      "ts",
      "tsx",
      "bmp",
      "gif",
      "jpg",
      "jpeg",
      "png",
      "psd",
      "svg",
      "webp",
      "m4v",
      "mov",
      "mp4",
      "mpeg",
      "mpg",
      "webm",
      "aac",
      "aiff",
      "caf",
      "m4a",
      "mp3",
      "wav",
      "html",
      "pdf",
      "otf",
      "ttf",
    ],
    "searchIgnore": [Function],
    "sourceExts": Array [
      "js",
      "json",
      "ts",
      "tsx",
    ],
  },
  "userConfig": Object {
    "assetsOutput": "/tmp/root/output/assets/by/customConfig/",
    "dev": false,
    "entryFile": "/tmp/root/index.js",
    "output": "/tmp/root/output/index.platform.bundle",
    "platform": "ios",
    "projectRoot": "/tmp/root",
    "split": true,
  },
}
`;
